frontend/src/components/HospitalInventoryEdit.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useEffect, useState } from 'react'
+import Col from 'react-bootstrap/Col'
+import Form from 'react-bootstrap/Form'
+import Card from 'react-bootstrap/Card'
+import Button from 'react-bootstrap/Button'
+import { useParams } from 'react-router-dom'
+import style from './HospitalInventoryEdit.module.css'
+import { HospitalInventory, HospitalAddress, fetchHospitalInventory, fetchHospitalAddress } from '../apis/infomed'
+
+
+interface HospitalInventoryParams {
+  hospitalId: string
+}
+
+export const HospitalInventoryEdit = ({hospitalId}: HospitalInventoryParams) => {
+
+  const [inventoryList, setInventoryList] = useState<HospitalInventory[]>([]);
+  const [hospitalAddress, setHospitalAddress] = useState<HospitalAddress[]>([]);
+
+  useEffect(() => {
+    (async () => {
+      const list = await fetchHospitalInventory(hospitalId);
+      setInventoryList(list);
+      
+      const address = await fetchHospitalAddress(hospitalId);
+      setHospitalAddress(address);
+    })();
+  }, [hospitalId]);
+
+  const handleInventoryChange  = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const updateList = [...inventoryList];
+    const target = e.target.getAttribute("data-index");
+    const dataIndex = parseInt(target !== null ? target : '0');
+    updateList[dataIndex]["available"] = parseInt(e.currentTarget.value);
```

Comment:
Note: by using a dot notation instead of brackets, you should get stronger typing (unless TypeScript compiler is smarter than I think): `updateList[dataIndex].available`